src/io/vss_store.rs

```diff
@@ -491,13 +502,19 @@ impl VssStoreInner {
 			"remove",
 		)?;
 
+		let obfuscated_key =
+			self.build_obfuscated_key(&primary_namespace, &secondary_namespace, &key);
+
+		let key_value = KeyValue { key: obfuscated_key, version: -1, value: vec![] };
```

Comment:
Can't we store just keys here?

Reply:
Currently we might get away with it, but going forward we might utilize the `version` field there, so I'd prefer to do the future-safer thing here.

Reply:
You say 'might'. Do you mean that just using keys can already be a problem without the version field?

Even if we use the version field in the future, `value` still isn't needed?

In general, I would avoid dead code (or fields in this case). It just raises questions with devs. At least it did with me. Or otherwise clearly document.

Final question - is a version needed for lazy deletes?

Reply:
We could map the key to `KeyValue` at a later point, and not use it internally? Not a blocker.

>Well, if we issue a delete on a certain version, the service could in the future ignore a delete if it already has a higher version, no? I.e., it would thereby detect that the value has been updated since the delete was issued.

I thought we didn't want rewrites of the same key after it has been deleted lazily, but not 100% sure anymore. There's no version currently anyway, so a worry for later.